server/src/main/scala/com/cloudera/livy/server/SessionServlet.scala

```diff
@@ -60,54 +65,45 @@ abstract class SessionServlet[S <: Session, R: ClassTag](sessionManager: Session
     Map(
       "from" -> from,
       "total" -> sessionManager.size(),
-      "sessions" -> sessions.view(from, from + size).map(clientSessionView)
+      "sessions" -> sessions.view(from, from + size).map(clientSessionView(_, request))
     )
   }
 
   val getSession = jget("/:id") {
-    val id = params("id").toInt
-
-    sessionManager.get(id) match {
-      case None => NotFound("session not found")
-      case Some(session) => clientSessionView(session)
+    withUnprotectedSession { session =>
```

Comment:
The behavior I decided on was:

- anyone can list sessions and read non-sensitive information about them. Meaning "/" and "/:id" routes don't need to be protected.
- anything that can show sensitive info (like job status) or modify a session needs to be protected.

We could decide that we don't want the first behavior; but then we might need some notion of a "superuser" that is allowed to look at the complete server state.